### PR TITLE
DEV: Move chat selection buttons out of chat-live-pane

### PR DESCRIPTION
**Pull request by martin-brennan**
At the moment there is just a Quote button when selecting messages,
but we are adding another button here and potentially more
in future, it makes more sense to have the chat selection
management buttons to be in their own component, not clogging
up chat-live-pane.

### CODE REVIEW

File: assets/javascripts/discourse/components/chat-selection-manager.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import { getOwner } from "discourse-common/lib/get-owner";
+import { action } from "@ember/object";
+import { ajax } from "discourse/lib/ajax";
+import { isTesting } from "discourse-common/config/environment";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import { schedule } from "@ember/runloop";
+import { inject as service } from "@ember/service";
+import getURL from "discourse-common/lib/get-url";
+
+export default Component.extend({
```
Comment (jjaffeux) [inline assets/javascripts/discourse/components/chat-selection-manager.js:13]: can you try using native class syntax please?

File: assets/javascripts/discourse/components/chat-selection-manager.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  tagName: "",
+  router: service(),
+  chatChannel: null,
+  selectedMessageIds: null,
+  showChatQuoteSuccess: false,
+  cancelSelecting: null,
+
+  @discourseComputed("selectedMessageIds")
```
Comment (jjaffeux) [inline assets/javascripts/discourse/components/chat-selection-manager.js:21]: you can compute on length directly, and we also have a gt() helper

File: assets/javascripts/discourse/components/chat-selection-manager.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        openOpts,
+      });
+    }
+  },
+
+  _showCopyQuoteSuccess() {
+    this.set("showChatQuoteSuccess", true);
+
+    schedule("afterRender", () => {
+      if (this._selfDeleted) {
```
Comment (jjaffeux) [inline assets/javascripts/discourse/components/chat-selection-manager.js:97]: where is this defined?

File: assets/javascripts/discourse/templates/components/chat-selection-manager.hbs [hunk 4]
```diff
@@ -0,0 +1,25 @@
+<div class="chat-selection-management">
```
Comment (jjaffeux) [inline assets/javascripts/discourse/templates/components/chat-selection-manager.hbs:1]: is there any css associated with it?

File: assets/javascripts/discourse/components/chat-selection-manager.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+  showChatQuoteSuccess: false,
+  cancelSelecting: null,
+
+  @discourseComputed("selectedMessageIds")
+  anyMessagesSelected(selectedMessageIds) {
+    return selectedMessageIds.length > 0;
+  },
+
+  @action
+  async quoteMessages() {
```
Comment (jjaffeux) [inline assets/javascripts/discourse/components/chat-selection-manager.js:27]: The general complexity of this action is very high, maybe we could split a few concerns

File: assets/javascripts/discourse/templates/components/chat-selection-manager.hbs (hunk 4 above)
Comment (martin-brennan) [reply assets/javascripts/discourse/templates/components/chat-selection-manager.hbs:1]: Yes there is mainly around the buttons and the success message shown on copy quote to clipboard

File: assets/javascripts/discourse/components/chat-selection-manager.js (hunk 3 above)
Comment (martin-brennan) [reply assets/javascripts/discourse/components/chat-selection-manager.js:97]: Hmm weird it's from the live pane, will get rid of it, not really sure what it's for.

File: assets/javascripts/discourse/components/chat-selection-manager.js (hunk 2 above)
Comment (martin-brennan) [reply assets/javascripts/discourse/components/chat-selection-manager.js:21]: Nice will change

File: assets/javascripts/discourse/components/chat-selection-manager.js (hunk 1 above)
Comment (martin-brennan) [reply assets/javascripts/discourse/components/chat-selection-manager.js:13]: Yep will do, just copied from the  live pane.

File: assets/javascripts/discourse/components/chat-selection-manager.js (hunk 5 above)
Comment (martin-brennan) [reply assets/javascripts/discourse/components/chat-selection-manager.js:27]: Sure I could probably split it up into more functions, one for being isolated, one for mobile etc.

File: assets/javascripts/discourse/templates/components/chat-selection-manager.hbs (hunk 4 above)
Comment (jjaffeux) [reply assets/javascripts/discourse/templates/components/chat-selection-manager.hbs:1]: So maybe we create an associated css files and move stuff related to this inside please

File: assets/javascripts/discourse/components/chat-selection-manager.js (hunk 2 above)
Comment (martin-brennan) [reply assets/javascripts/discourse/components/chat-selection-manager.js]: Tried to use `gt` and `discourseComputed` on the length, and neither updated when the selected messages changed in chat live pane. Ended up with the normal ember `@computed` property:

```javascript
  @computed("selectedMessageIds.length")
  get anyMessagesSelected() {
    return this.selectedMessageIds.length > 0;
  }
```

**Review by tgxworld — APPROVED** [review]

**Pull request merged**